.github/workflows/build-and-test.yml

```diff
@@ -289,6 +289,13 @@ jobs:
         with:
           name: CoverageReport # Artifact name
           path: coveragereport # Directory containing files to upload
+      
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v5
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
Can you also send a PR for the internal docs repo (I'll send the link) with instructions for how to regenerate this?